asset-transfer-basic/chaincode-java/src/main/java/org/hyperledger/fabric/samples/assettransfer/ContractMain.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.fabric.samples.assettransfer;
+
+import org.hyperledger.fabric.contract.ContractRouter;
+
+public final class ContractMain {
+
+    private ContractMain() {
+    }
+
+    public static void main(final String[] args) throws Exception {
```

Comment:
I'm suspicious of this main method since the Java chaincode was updated so that CCaaS doesn't require a main method- it feels like an anti-pattern which probably shouldn't be in the sample? cc @mbwhite

Reply:
I'm neutral on this one.  I tried both techniques, and the one that felt "correct" was to include a Main driver in the sample - merely for the convenience of setting up run targets in Eclipse and IntelliJ.   In these tools there are some functions that will traverse / reflect methods in the **local** / **active** package to find runnable methods with a main driver.  Finding the main function in a dependent library is a little bit of work.  While I prefer a bit of magic behind the scenes, I preferred the explicit entrypoint for clarity. 

That said, I'm OK with either approach.  The main driver is called out in the build.gradle, so it shouldn't be a complete mystery to track down the run target / main class.   I changed the build.gradle `mainClass` to align with the local class.

Matthew do you have a preference here?  I'm happy to switch the approach either way.

Reply:
I'm culling this.  Back to the chaincode router main().... 

- The entrypoint is called out in build.gradle.  It's OK to assume a minimum level Java / gradle familiarity, including tracking down entry points, SDKs, dependencies, and the like.  If an end-user wants to implement a Main() - OK. 

- Removing the `main()` suggests that a chaincode driver can and should function in EITHER mode - CCaaS or the legacy build pipeline.   Introducing a main() in the sample lessens the flexibility as it suggests that CCaaS is the ONLY target runtime.